src/ContentRepository/Linq/SnExpression.cs

```diff
@@ -100,6 +100,7 @@ private static LucQuery BuildLucQuery(System.Linq.Expressions.Expression express
                     query.Top = childrenDef.Top;
                 if (childrenDef.Sort != null)
                     query.SetSort(childrenDef.Sort);
+                query.AllPages = childrenDef.AllPages;

```

Comment:
We override the value here unconditionally, maybe we should use a nullable value or some other solution, because the other values here (sort, top, etc) are overridden only if changed/non-default.